docs/site/pages/plugins/coroutines.mdx

```diff
@@ -0,0 +1,112 @@
+---
+title: Coroutines
+platform: android
+---
+
+# Kotlin Coroutines
+
+Kotlin coroutines is a popular Kotlin library to handle asynchronous programming and concurrency in the JVM. More information on Coroutines can be found [here](https://kotlinlang.org/docs/coroutines-overview.html)
+This module contains several JVM exclusive plugins to aid various tooling needs.
+
+## UpdatesPlugin
+
+Plugin using a `ReceiveChannel` to provide an updated asset whenever the Player View is updated.
+
+### Usage
+
+<PlatformTabs>
+  <android>
+In build.gradle
+
+```kotlin
+implementation "com.intuit.player.plugins:coroutines:$JVM_PLAYER_VERSION"
+```
+
+In Player constructor
+```kotlin
+import com.intuit.player.plugins.coroutines.UpdatesPlugin
+
+val plugins = listOf(UpdatesPlugin())
+AndroidPlayer(plugins)
+```
+  </android>
+</PlatformTabs>
+
+
+### API
+
+<PlatformTabs>
+  <android>
+The API for this plugin revolves around the `ReceiveChannel` for asset updates.
+
+```kotlin
+// Plugin API
+
+/** Public property that is used by the [UpdatesPlugin] to send [Asset] updates, can be used in a CoroutineContext to receive [Asset] updates. */
+public val updates: ReceiveChannel&lt;Asset?&gt;
+
+/** This blocks the current thread until a new update is received or the timeout specified has passed. */
+public fun waitForUpdates(timeout: Long = 500): Asset?
+
+// Experimental
+
+/** This cleans out the [updates] channel to prepare it to receive new updates */
+public fun flush(): Unit
+
+// Player extensions
+public val Player.updatesPlugin: UpdatesPlugin?
+public fun Player.waitForUpdates(timeout: Long = 500): Asset?
+```
```

Comment:
Same comment around just linking to source - if we want separate API docs, we should just generate them